### PR TITLE
Enable testing to verify that favorite sessions are displayed on the favorite screen

### LINKED ISSUE #343: Add some test cases to verify that favorite sessions are displayed correctly on the  favorites screen
Currently, we can only check without any favorites. As mentioned in [the comments](https://github.com/DroidKaigi/conference-app-2024/pull/248#discussion_r1712951458), by injecting test data into UserDataStore, we should be able to test screens with favorite sessions.

### PR DESCRIPTION
**Pull request by Yamada-Ika**
## Issue
- close #343

## Overview (Required)
- I have enhanced the UI tests for the favorites screen. Previously, only the state with no favorites was tested, but now it is also possible to test the state with favorites.

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :--:
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="" width="300" >

### CODE REVIEW

File: feature/favorites/src/androidUnitTest/kotlin/io/github/droidkaigi/confsched/favorites/FavoritesScreenTest.kt [hunk 1]
```diff
@@ -27,9 +27,11 @@ class FavoritesScreenTest(
     @Inject
     lateinit var favoritesScreenRobot: FavoritesScreenRobot
 
+    @Inject lateinit var testDispatcher: TestDispatcher
+
     @Test
     fun runTest() {
-        runRobot(favoritesScreenRobot) {
+        runRobot(context = testDispatcher, robot = favoritesScreenRobot) {
```
Comment (Yamada-Ika) [inline feature/favorites/src/androidUnitTest/kotlin/io/github/droidkaigi/confsched/favorites/FavoritesScreenTest.kt:34]: If don't pass the TestDispatcher injected by Hilt, the TestDispatcher used in the tests will differ from the one passed to UserDataStore (passed [here](https://github.com/DroidKaigi/conference-app-2024/blob/a6f1fbfa060110e97082dbaa3a926412cc101d77/core/testing/src/main/java/io/github/droidkaigi/confsched/testing/data/TestDataStoreModule.kt#L30)), A runtime error occured. 

So, I have modified the `runRobot` interface to allow for changing the dispatcher. Is this acceptable?

File: core/testing/src/main/java/io/github/droidkaigi/confsched/testing/robot/FavoritesScreenRobot.kt [hunk 2]
```diff
@@ -21,11 +30,39 @@ class FavoritesScreenRobot @Inject constructor(
                 )
             }
         }
+        waitUntilIdle()
+    }
+
+    suspend fun setupFavoriteSession() {
+        userDataStore.toggleFavorite(TimetableItemId("0570556a-8a53-49d6-916c-26ff85635d86"))
```
Comment (Yamada-Ika) [inline core/testing/src/main/java/io/github/droidkaigi/confsched/testing/robot/FavoritesScreenRobot.kt:37]: I wanted to add test cases for changing the filter in a situation where there are multiple favorites.
However, due to the following reasons, I am currently reproducing a scenario with only one favorite.

To use the favorite data in the test, it is necessary to pass an id to `UserDataStore#toggleFavorite` that matches the id of the `SessionResponse`.  Most of the fake data for `SessionResponse` used in the tests includes commas in the id, which prevents the correct generation of favorite data. (In `UserDataStore`, ids are stored in preferences separated by commas, so if the id contains commas the same value cannot to be retrieved upon saving.)

Therefore, I modified the `SessionResponse` ids to be sequential number. However, this caused the `TimetableItemDetailScreenTest` tests to fail.

So, for now, I hard-coded ids without commas to ensure that the test for the existence of favorites can be executed.

File: feature/favorites/src/androidUnitTest/kotlin/io/github/droidkaigi/confsched/favorites/FavoritesScreenTest.kt (hunk 1 above)
Comment (takahirom) [reply feature/favorites/src/androidUnitTest/kotlin/io/github/droidkaigi/confsched/favorites/FavoritesScreenTest.kt]: I injected the testDispatcher in DefaultScreenRobot.

File: core/testing/src/main/java/io/github/droidkaigi/confsched/testing/robot/FavoritesScreenRobot.kt (hunk 2 above)
Comment (takahirom) [reply core/testing/src/main/java/io/github/droidkaigi/confsched/testing/robot/FavoritesScreenRobot.kt]: I fixed the problem. The cause of the error was the filter logic in the Repository. So I modified it to always use this logic.

**Comment by takahirom** [comment]
@Corvus400 @YusukeMoriJapan I apologize for the delay. I'll merge this shortly.

**Review by takahirom — APPROVED** [review]
Thank you for letting me know the problem and contribution!

**Pull request merged**